### PR TITLE
More eclipse compiler fixes.

### PR DESCRIPTION
**Pull request by hgschmie**
Compilation errors flagged by the Eclipse compiler. Those can be
reproduced inside IDEA by following
http://blog.dripstat.com/post/103658392234/why-you-should-use-the-eclipse-compiler-in

### CODE REVIEW

File: presto-cassandra/src/main/java/com/facebook/presto/cassandra/CachingCassandraSchemaProvider.java [hunk 1]
```diff
@@ -176,7 +179,7 @@ public void flushCache()
         return ImmutableList.copyOf(getCacheValue(schemaNamesCache, "", RuntimeException.class).keySet());
     }
 
-    private Map<String, String> loadAllSchemas()
+    private ImmutableMap<String, String> loadAllSchemas()
```
Comment (dain) [inline presto-cassandra/src/main/java/com/facebook/presto/cassandra/CachingCassandraSchemaProvider.java]: Is this change needed?  Normally, we just return the interface type.

File: presto-main/src/main/java/com/facebook/presto/operator/SettableLookupSourceSupplier.java [hunk 2]
```diff
@@ -46,10 +46,12 @@ public SettableLookupSourceSupplier(List<Type> types)
         return types;
     }
 
+    @SuppressWarnings("RedundantCast")
     @Override
     public ListenableFuture<LookupSource> getLookupSource(OperatorContext operatorContext)
     {
-        return transform(lookupSourceFuture, (AsyncFunction<SharedLookupSource, LookupSource>) Futures::immediateFuture);
+        // The Eclipse compiler requires the explicit <LookupSource> cast.
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/operator/SettableLookupSourceSupplier.java]: "the explicit ... cast" -> "the explicit ... type argument", right?

File: presto-main/src/main/java/com/facebook/presto/operator/SettableLookupSourceSupplier.java [hunk 3]
```diff
@@ -46,10 +46,12 @@ public SettableLookupSourceSupplier(List<Type> types)
...
+        return transform(lookupSourceFuture, (AsyncFunction<SharedLookupSource, LookupSource>) Futures::<LookupSource>immediateFuture);
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/operator/SettableLookupSourceSupplier.java]: IDEA thinks the cast is redundant after adding the type argument (hence the suppression you added). Without it, the Eclipse compiler seems to crash:
```
Error:java: Internal compiler error: java.lang.NullPointerException at org.eclipse.jdt.internal.compiler.lookup.InferenceContext18.checkExpression(InferenceContext18.java:796)
```
This sees to be https://bugs.eclipse.org/bugs/show_bug.cgi?id=438945

Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/operator/SettableLookupSourceSupplier.java]: That bug seems to be fixed in 4.4.2. I see this in the dependency for plexus-compiler-eclipse:
```
<groupId>org.eclipse.tycho</groupId>
<artifactId>org.eclipse.jdt.core</artifactId>
<version>3.10.0.v20140604-1726</version>
```
Does this mean it is using a very old version? cc @jvanzyl

**Comment by hgschmie** [comment]
I would hold off applying this pull request for now. The upcoming Eclipse Mars compiler seems to have all the problems fixed (it compiles my test suite without errors) but in turn introduced a new error (see https://bugs.eclipse.org/bugs/show_bug.cgi?id=469317)

**Comment by hgschmie** [comment]
Downloading Eclipse Mars and loading Presto does indeed work and has no errors (only warnings).

File: presto-cassandra/src/main/java/com/facebook/presto/cassandra/CachingCassandraSchemaProvider.java (hunk 1 above)
Comment (hgschmie) [inline presto-cassandra/src/main/java/com/facebook/presto/cassandra/CachingCassandraSchemaProvider.java]: It is required for the Luna compiler; it seems to be a bug, the Mars compiler understands this correctly.

**Comment by martint** [comment]
Should we close this @hgschmie ?
